dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/validation/RorValidator.java

```diff
@@ -1,46 +1,34 @@
-package edu.harvard.iq.dataverse.validation.field.validators;
+package edu.harvard.iq.dataverse.validation;
 
-import edu.harvard.iq.dataverse.common.BundleUtil;
-import edu.harvard.iq.dataverse.persistence.dataset.ValidatableField;
 import edu.harvard.iq.dataverse.validation.field.ValidationResult;
 import org.apache.commons.lang3.StringUtils;
-import org.omnifaces.cdi.Eager;
 
-import javax.enterprise.context.ApplicationScoped;
+import javax.ejb.Stateless;
 import java.util.Arrays;
 import java.util.Collections;
-import java.util.List;
 import java.util.Map;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
-@Eager
-@ApplicationScoped
-public class RorValidator extends MultiValueValidatorBase {
-
+@Stateless
+public class RorValidator {
+    public static final String INVALID_FORMAT_ERROR_CODE = "ror.invalid.format";
+    public static final String INVALID_CHECKSUM_ERROR_CODE = "ror.invalid.checksum";
     private static final Map<String, Integer> DECODE_SYMBOL_VALUES = Initializer.initializeDecodeSymbolValues();
 
     // -------------------- LOGIC --------------------
 
-    @Override
-    public String getName() {
-        return "ror_validator";
-    }
-
-    @Override
-    public ValidationResult validateValue(String fullRor, ValidatableField field, Map<String, Object> params,
-                                          Map<String, ? extends List<? extends ValidatableField>> fieldIndex) {
-        String fieldName = field.getDatasetFieldType().getDisplayName();
+    public ValidationResult validate(String fullRor) {
         if (StringUtils.isBlank(fullRor)
             || !fullRor.matches("https://ror\\.org/0[a-hjkmnp-tv-z0-9]{6}[0-9]{2}")) {
-            return ValidationResult.invalid(field, BundleUtil.getStringFromBundle("ror.invalid.format", fieldName));
+            return ValidationResult.invalid(INVALID_FORMAT_ERROR_CODE);
         }
         String value = fullRor.substring(fullRor.lastIndexOf("/") + 1);
         String encoded = value.substring(0, 7);
         long checksum = Long.parseLong(value.substring(7));
         return checksum == computeChecksum(encoded)
                 ? ValidationResult.ok()
-                : ValidationResult.invalid(field, BundleUtil.getStringFromBundle("ror.invalid.checksum", fieldName));
+                : ValidationResult.invalid(INVALID_CHECKSUM_ERROR_CODE);
```

Comment:
I am wondering how much we want to bind both `classic jsf validation` and `dataset field validation` mechanisms. And I lean towards the option that this need to be two separate things. That would mean that we shouldn't use `ValidationResult` as a return type here (which is specific to `dataset field validation`) but for example use some kind of enum (maybe `enum RorValidationResult { OK, INVALID_FORMAT, INVALID_CHECKSUM }`).

What do you think?

Reply:
Yeah, it's not ideal. Forgot to add a note about his in the PR description. So, initially I did actually add an enum but that resulted in repetitive if/else or switch checking in multiple places (for both ror and orcid validators in `BaseUserPage` and in the field validator).
Then I thought that well maybe I could rename `ValidationResult` to `FieldValidationResult` and let it extend a new `ValidationResult`. That resulted in a lot of changed files and would be very distracting in a review.
If you would also agree with that approach, I would propose to have that refactoring be done in isolation in a separate PR.

Reply:
I've created the refactor PR: https://github.com/CeON/dataverse/pull/2454